sky/optimizer.py

```diff
@@ -894,13 +1014,18 @@ def sort_key(row, accelerator_spot_list=accelerator_spot_list):
 
                 rows = sorted(rows, key=sort_key)
             else:
-                rows = sorted(rows, key=lambda row: float(row.cost_str))
+                rows = sorted(
```

Comment:
When there are clouds with `Unknown` footprint, the optimizer selects that cloud. Is this behavior expected? Should we instead pick the lowest across all non-unknown values?
 
<img width="1292" alt="image" src="https://github.com/skypilot-org/skypilot/assets/4416605/e426c000-2128-4d0f-9fd2-ddb570c9aa6f">

Reply:
Got it.  I was able to recreate with a local `k8s`.  I'm on it!

Reply:
Fixed.